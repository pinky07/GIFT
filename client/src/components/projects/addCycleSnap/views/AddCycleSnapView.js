import React from 'react';

import Heading from 'grommet/components/Heading';
import Form from 'grommet/components/Form';
import FormFields from 'grommet/components/FormFields';
import FormField from 'grommet/components/FormField';
import TextInput from 'grommet/components/TextInput';
import Box from 'grommet/components/Box';
import Footer from 'grommet/components/Footer';
import Button from 'grommet/components/Button';
import NumberInput from 'grommet/components/NumberInput';
import DateTime from 'grommet/components/DateTime';
import Layer from 'grommet/components/Layer';
import CheckBox from 'grommet/components/CheckBox';

const AddCycleSnapView = ({ viewModel }) => {
  const { errors } = viewModel;
  const { formCallbacks } = viewModel;
  const { projectName } = viewModel;
  const { cycleSnapName } = viewModel;
  const { startDate } = viewModel;
  const { endDate } = viewModel;
  const { targetedPoints } = viewModel;
  const { achievedPoints } = viewModel;
  const { isMoodAvailable} = viewModel;
  const { mood } = viewModel;

  return <Layer align='center' closer={true} onClose={formCallbacks.onClose}>
    <Box pad={{ vertical: 'large', horizontal: 'small' }}>

      <Heading>Add cycle snap</Heading>
      <h2>Project: {projectName}</h2>

      <Form compact={false}>
        <FormFields>
          <fieldset>
            <FormField label='Cycle name or id' htmlFor="cycleNameInput" error={errors.name}>
              <TextInput id='cycleNameInput' onDOMChange={formCallbacks.onNameChange} value={cycleSnapName} />
            </FormField>

            <FormField label='Start Date' htmlFor="cycleStartDate" error={errors.startDate}>
              <DateTime id='cycleStartDate' format='YYYY-MM-DD' value={startDate} onChange={formCallbacks.onStartDateChange} />
            </FormField>

            <FormField label='End Date' htmlFor="cycleEndDate" error={errors.endDate}>
              <DateTime id='cycleEndDate' format='YYYY-MM-DD' value={endDate} onChange={formCallbacks.onEndDateChange} />
            </FormField>

            <FormField label='Targeted points' htmlFor='targetedPointsId' error={errors.targetedPoints}>
              <NumberInput id='targetedPointsId' step={1} value={targetedPoints} min={0} onChange={formCallbacks.onTargetedPointsChange} />
            </FormField>

            <FormField label='Achieved points' htmlFor='achievedPointsId' error={errors.achievedPoints}>
              <NumberInput id='achievedPoints' step={1} value={achievedPoints} min={0} onChange={formCallbacks.onAchievedPointsChange} />
            </FormField>

            <FormField>
              <CheckBox label='Is Mood Available?'
                        reverse={true}
                        toggle={true}
                        disabled={false}
                        value={isMoodAvailable}
                        onChange={formCallbacks.onIsMoodAvailableChange}/>
            </FormField>

            <FormField label='Mood' htmlFor='moodId' error={errors.mood}>
              <NumberInput id='mood' step={0.01} min={1} max={3} value={mood} onChange={formCallbacks.onMoodChange} />
            </FormField>

          </fieldset>
        </FormFields>

        <Footer pad={{ "vertical": "medium" }}>
          <Box>
            <Button label='Add' type='submit' primary={true} onClick={formCallbacks.onClick} />
          </Box>
          <Box pad={{ horizontal: 'small' }}>
            <Button label='Cancel' secondary={true} onClick={formCallbacks.onClose} />
          </Box>
        </Footer>
      </Form>
    </Box>
  </Layer>
}

export default AddCycleSnapView;