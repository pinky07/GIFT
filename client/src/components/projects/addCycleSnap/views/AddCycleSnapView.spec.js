import React from 'react';
import renderer from 'react-test-renderer';

import AddCycleSnapView from './AddCycleSnapView';

describe('AddCycleSnapView', () => {

    let actual;

    it('should render the correct data values and validation errors', () => {
        let viewModel = getViewModelWithErrors();
        actual = renderer.create(<AddCycleSnapView viewModel={viewModel} />).toJSON();

        expect(actual).toMatchSnapshot();
    });

    function getViewModelWithErrors() {
        return {
            projectId: '7',
            projectName: 'Exceptional Project',
            cycleSnapName: '1',
            startDate: '2017-01-02',
            endDate: '2017-01-15',
            targetedPoints: '8',
            achievedPoints: '13',
            errors: {
                name: 'error name',
                startDate: 'error startDate',
                endDate: 'error endDate',
                targetedPoints: 'error targetedPoints',
                achievedPoints: 'error achievedPoints'
            },
            formCallbacks: {
                onNameChange: jest.fn(),
                onStartDateChange: jest.fn(),
                onEndDateChange: jest.fn(),
                onTargetedPointsChange: jest.fn(),
                onAchievedPointsChange: jest.fn(),
                onClick: jest.fn(),
                onClose: jest.fn()
            }
        };
    }
});