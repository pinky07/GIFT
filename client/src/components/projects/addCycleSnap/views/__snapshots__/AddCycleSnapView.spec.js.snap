// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCycleSnapView should render the correct data values and validation errors 1`] = `
<span
  style={
    Object {
      "display": "none",
    }
  }
/>
`;
